reader/components/pdf-toolbar/PDFToolbar.jsx

```diff
@@ -29,20 +38,78 @@ const PDFToolbar = ({ viewer, eventBus }) => {
 
   useEffect(() => {
     // component didMount
-    const onPageChange = e => currPageChange(e.pageNumber)
+    const setTimeoutVisibility = () =>
+      setTimeout(() => {
+        changeVisibility(false)
+      }, 1000)
+
+    let timeoutId
+
+    const onPageChange = e => {
+      if (ui.isRelatedPapersScrolled && timeoutId)
+        dispatch(unsetRelatedPapers())
+      if (timeoutId) clearTimeout(timeoutId)
+      timeoutId = setTimeoutVisibility()
+      changeVisibility(true)
+
+      currPageChange(e.pageNumber)
+    }
     eventBus.on('pagechanging', onPageChange)
 
     // component will unmount
     return () => eventBus.off('pagechanging', onPageChange)
   }, [])
 
+  useEffect(() => {
+    if (relatedPapersClicked === null) return
+
+    if (relatedPapersClicked) {
+      setReadPosition(toolbarRef.current.parentNode.scrollTop)
+      dispatch(scrollToRelatedPapers())
+    } else {
+      toolbarRef.current.parentNode.scroll({
+        left: 0,
+        top: readPosition + 1000,
+        behavior: 'auto',
+      })
+
+      // needs to be within timeout otherwise events
+      // get merged and everything is scrolled smoothly not only last 900px
+      setTimeout(
+        () =>
+          toolbarRef.current.parentNode.scroll({
+            left: 0,
+            top: readPosition,
+            behavior: 'smooth',
+          }),
+        50
+      )
+      dispatch(unsetRelatedPapers())
+    }
+  }, [relatedPapersClicked])
+
   return (
-    <div className="pdf-toolbar d-flex flex-row justify-content-end align-items-center">
-      <div className="pdf-preferences d-flex flex-row align-items-center justify-content-between mr-5">
+    <div
+      className={`pdf-toolbar d-flex flex-wrap justify-content-end align-items-center ${
+        isVisible ? 'pdf-toolbar-visible' : ''
+      }`}
+      ref={toolbarRef}
+    >
+      <div className="pdf-related-papers d-flex flex-row align-items-center justify-content-between mr-lg-5 mr-2 order-3">
+        <button
+          title="Show related papers"
+          type="button"
+          className="btn m-auto"
+          onClick={() => setRelatedPapersClicked(!relatedPapersClicked)}
+        >
+          {!relatedPapersClicked ? 'Related papers' : 'Back to reading'}
+        </button>
+      </div>
```

Comment:
I would like to slightly change this but I am not sure in which way. Basically, I like your solution, it's quite creative but I am not sure I would like to have this button as a part of the interface.

I suggest to fix the padding of the button and keep it as it is.

<img width="214" alt="" src="https://user-images.githubusercontent.com/8440244/68583605-3f28f900-0486-11ea-80a5-8730b61cbc21.png">

Reply:
https://github.com/oacore/reader/pull/89/commits/bf8d8a6d02c3004eda34f43ab864585a55dfdf99 I believe you meant it this way and not changing the height of the whole element (it looked ugly).

Reply:
Yes, that is what I meant. Thank you